paths/git-trouble/fc-wrongbranch.md

```diff
@@ -0,0 +1,67 @@
+---
+layout: simple-class
+header:
+  overlay_image: cover.jpeg
+  overlay_filter: rgba(46, 129, 200, 0.6)
+title: Committed to the Wrong Branch!
+permalink: /git-trouble/05
+next-page: /git-trouble/02
+facilitator: false
+sidebar:
+  nav: "advanced"
+main-content: |  
+
+  We have all been there, cruising along, making changes, making commits, and just as you are about to call it a night, you realize you just commited all of your changes to `Master` and not that new branch you forgot to `checkout` to.
+
+  Fear not! You can salvage those changes and put them where they belong! Next time, try to remember to run a quick `git checkout BRANCH` before you get to working on that sweet Pull Request :wink:.
+
+pushed: |
+    When trying to fix committing to the wrong branch, it is important to identify if you need to fix 1 commit or multiple commits before continuing.
+
+    ## Single Commit
+
+    1. Ensure you are on the correct branch and enter: `git log --oneline`. Identify the SHA-1 hash for the **adding file 3** commit.
+    1. Enter: `git revert SHA-1`, where SHA-1 is the hash for the **adding file 3** commit. You can modify the revert commit message if you would like or just close the editor.
+    1. Create a new branch by entering: `git checkout -b NEWBRANCH`, where NEWBRANCH is the name of your new branch.
+    1. Enter: `git reflog`,
+    1. On the new branch, enter: `git cherry-pick SHA-1`, where SHA-1 is the hash for the **adding file 3** commit.
+
+    ## Multiple Commits
+
+    1. Ensure you are on the correct branch and enter: `git log --oneline`. Identify the SHA-1 hash for the commit where **file 1** was added.
+    1. Enter `git reset --soft SHA-1`, where SHA-1 is the commit where **file 1** was added.
+    1. Enter: `git checkout BRANCH`, where BRANCH is the name of the branch you should have made the commits to. If you don't have an existing branch, enter: `git checkout -b BRANCH`, where BRANCH is the name of the new branch you want to create.
+    1. Enter: `git status`, files 2 - 6 should be in the Staging Area.
+    1. Enter: `git commit -m "MESSAGE"`, where message is your new commit message, for example `Add files 2 through 6` could work.
+    1. Enter: `git push`. If you haven't pushed from this branch before, you will be prompted by Git to set the `upstream` for the branch. To set the upstream, enter: `git push -u origin new`.
+    1. Enter: `git checkout BRANCH`, where BRANCH is the branch you removed your errant commits from.
+    1. Enter `git push`.  
```

Comment:
Since we've altered history that has been pushed, I believe this would also required a force push.